### ISSUE TITLE
Printing secured service with Print widget produces error with code :400

### ISSUE THREAD

**Issue by nagmayasmin** [Esri/resource-proxy#222]
The developer's sample located at 
Print templates with esri.request
"https://developers.arcgis.com/javascript/jssamples/widget_print_esri_request.html" has been replaced with secured service "http://sampleserver6.arcgisonline.com/arcgis/rest/services/USA_secure_user1/MapServer/0" with the proper configuration of proxy.config file to print. The secured service is displayed correctly, but the print doesn't work and erros out in developer tools with "unable to complet operation, code:400". 

One thing to notice, if we don't add the secured service in proxy.config file, the application prompts for credential. After putting the credential, I am able to see secured service and able to print out without any issue.

A bug has already been submitted regarding this issue:
[BUG-000086706:Printing secured map service with Print widget produces error "Unable to complete operation, code:400"]

**Comment by jgravois** [comment]
have you reviewed the documentation on printing secured services [here](http://resources.arcgis.com/en/help/main/10.2/#/Printing_maps_that_contain_secured_services/0154000005q3000000/)?

**Comment by afili** [comment]
this basically describes about storing the credentials in the print service, right? i remember that we tested that in the past but it still did not solve the issue when using a proxy. but we will test again to double check. but regardless of storing the credentials, we see that the token is not passed in the export_webmap_json when using the proxy, while it is getting passed when using the Identity Manager. I know that the proxy will not currently handle this, not sure if it is something we can support since we would have to parse the export_webmap_json.

**Comment by jgravois** [comment]
i haven't tested it in quite awhile, but i was under the impression that creating the custom print service was the key to supporting printing of secured services and that it doesn't matter whether or not your REST request contains a token because, as you said, the credentials are stored within the print service itself.

all that being said, i could be mistaken.

**Comment by tr3vorm** [comment]
I use secured print services with the proxy. The secure print service has credentials stored to access secure layers, but a token still has to be passed to the print service (GPTool) assuming that is secured at that level too. I recall a relevant bug being fixed in ArcGIS 10.3 - not very helpful sorry.

**Comment by hgonzago** [comment]
You may want to take a look at http://resources.arcgis.com/en/help/flex-viewer/concepts/index.html#/Printing_secure_services/01m30000005s000000/.

It's an older doc that was made for the Flex Viewer, but the premise should still apply. Specifically the section that reads, "2. Secondly, the print service needs to be able to access the secure layer(s) as well. Essentially, you are trying to access the secure layer(s) from within the secured print service." and the information given below this "When the print service sends its request, there is no header information associated with it. In other words, if you created your token using http referer for the client, it would not work since there is no header information containing this http referer. The only time that this would work is if the long-lived token was generated using an IP address of the actual server hosting the print service. Although this would work, it is highly unlikely to see a scenario such as this within a production environment."

Does this sound similar to the situation that you are experiencing?

**Comment by afili** [comment]
Thanks @hgonzago 

Yes, I think that the issue is that the print service is not able to access the secure layers. Not sure if it is related with the long-lived token, but we followed the steps in this doc http://resources.arcgis.com/en/help/main/10.2/#/Printing_maps_that_contain_secured_services/0154000005q3000000/ that @jgravois provided in order to provide the name and password to the service. 

After the credentials were stored in the print service, we were able to print a secured map service. However, we were still unable to print a secured feature layer which was the user's initial request. This was because we were using a 10.2 print service. This is related with the bug [#NIM094431 A custom print service with saved credentials created from the Export Web Map task fails to print individual feature layers from a secured map service. ] which is fixed in 10.3. We then tried with 10.3 and confirmed that it works.

So the users should be able to print a secured feature layer by using a 10.3 print service that has stored credentials.
